openstudiocore/src/model/ElectricLoadCenterInverterPVWatts.cpp

```diff
@@ -0,0 +1,215 @@
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  ElectricLoadCenterInverterPVWatts_Impl::ElectricLoadCenterInverterPVWatts_Impl(const IdfObject& idfObject,
+                                                                                 Model_Impl* model,
+                                                                                 bool keepHandle)
+    : Inverter_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == ElectricLoadCenterInverterPVWatts::iddObjectType());
+  }
+
+  ElectricLoadCenterInverterPVWatts_Impl::ElectricLoadCenterInverterPVWatts_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                                 Model_Impl* model,
+                                                                                 bool keepHandle)
+    : Inverter_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == ElectricLoadCenterInverterPVWatts::iddObjectType());
+  }
+
+  ElectricLoadCenterInverterPVWatts_Impl::ElectricLoadCenterInverterPVWatts_Impl(const ElectricLoadCenterInverterPVWatts_Impl& other,
+                                                                                 Model_Impl* model,
+                                                                                 bool keepHandle)
+    : Inverter_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& ElectricLoadCenterInverterPVWatts_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty())
+    {
+      result.push_back("Inverter DC to AC Efficiency");
+      result.push_back("Inverter DC Input Electric Power");
+      result.push_back("Inverter DC Input Electric Energy");
+      result.push_back("Inverter AC Output Electric Power");
+      result.push_back("Inverter AC Output Electric Energy");
+      result.push_back("Inverter Conversion Loss Power");
+      result.push_back("Inverter Conversion Loss Energy");
+      result.push_back("Inverter Conversion Loss Decrement Energy");
+      result.push_back("Inverter Thermal Loss Rate");
+      result.push_back("Inverter Thermal Loss Energy");
+      result.push_back("Inverter Ancillary AC Electric Power");
+      result.push_back("Inverter Ancillary AC Electric Energy");
+    }
+    return result;
+  }
+
+  IddObjectType ElectricLoadCenterInverterPVWatts_Impl::iddObjectType() const {
+    return ElectricLoadCenterInverterPVWatts::iddObjectType();
+  }
+
+  std::vector<ScheduleTypeKey> ElectricLoadCenterInverterPVWatts_Impl::getScheduleTypeKeys(const Schedule& schedule) const {
+    std::vector<ScheduleTypeKey> result;
+    return result;
+  }
+
+  boost::optional<ThermalZone> ElectricLoadCenterInverterPVWatts_Impl::thermalZone() const {
+    return boost::none;
+  }
+
+  bool ElectricLoadCenterInverterPVWatts_Impl::setThermalZone(const ThermalZone& thermalZone) {
+    return boost::none;
```

Comment:
Seems like the IDF object should have a 'Zone Name' field

Reply:
I believe PVWatts doesn't use zone at all. I'm redefining here because it's a virtual method defined in https://github.com/NREL/OpenStudio/blob/pvwatts/openstudiocore/src/model/Inverter.hpp#L60.

Reply:
Yeah but in E+ it should have a zone name

Reply:
Not your problem....